### DISCUSSION TITLE
Pick up globals declared in d.ts files

### DISCUSSION THREAD

**Discussion by quantizor** [biomejs/biome#283] [Ideas]
It seems like Biome doesn't resolve tsconfig includes and thus pick up on globals defined in custom `d.ts` files and types supplied by the tsconfig `types` compiler option.

This leads to stuffing `javascript.globals` with all sorts of things that probably shouldn't be marked carte-blanche global.

This is a point of divergence that I think is actually a problem because it can lead to sneaky bugs when the global variable resolution algorithm differs significantly from TS's own.

**Comment by NexZhu** [comment]
+1 This is really PITA to list all globals
e.g., for writing Jest test, you need `describe`, `it`...

**Comment by dmythro** [comment]
Yes! We need this!

**Comment by xmlking** [comment]
SvelteKit projects has app.d.ts , my vscode always show them in red. 
need this ASAP

**Comment by quezak** [comment]
it would be great to have it supported!

As a workaround, here's a handy `overrides` config snippet for Jest globals, for those who stumble here from google (might need fine tuning for your usecase):
```jsonc
  "overrides": [
    {
      "include": ["*.spec.ts"],
      "javascript": {
        "globals": [
          "afterEach",
          "afterAll",
          "beforeEach",
          "beforeAll",
          "describe",
          "expect",
          "it",
          "test",
          "jest",
        ],
      },
    },
  ],
```

**Reply by Makentosh** [reply]
thank you! )

**Comment by naturalethic** [comment]
For those finding this because they have `noUndeclaredVariables` turned on for your js/ts.  You can turn this off specifically for ts with:
```json
"overrides": [
    {
      "include": ["**/*.ts", "**/*.tsx"],
      "linter": {
        "rules": {
          "correctness": {
            "noUndeclaredVariables": "off"
          }
        }
      }
    }
  ]
```
Then let ts lint its own undeclared variables, but the rule will remain on for javascript.

**Comment by Crauzer** [comment]
+1
Is there any plan to add support for this ?
